src/Raven.Studio/typescript/common/extensions.ts

```diff
@@ -191,9 +193,140 @@ class extensions {
             $(document).off("shown.bs.modal", onModalShown);
         });
     }
+    
+    private static verifyBindingLocation(bindingsArray: string[], binding: string, accessBindingLocation: number) {
+        const bindingLocation = bindingsArray.indexOf(binding);
 
+        if (bindingLocation > accessBindingLocation) {
+            throw new Error(`The '${binding}' binding must come BEFORE the 'requiredAccess' binding in your html.`);
+        }
+    }
+    
     private static installBindingHandlers() {
+        ko.bindingHandlers["requiredAccess"] = {
+            init: (element: Element,
+                   valueAccessor: KnockoutObservable<Raven.Client.ServerWide.Operations.Certificates.DatabaseAccess>,
+                   allBindings) => {
+                
+                const requiredAccessLevel = ko.unwrap(valueAccessor());
+                const levelTypes = ["Read", "ReadWrite", "Admin"];
+
+                if (!_.includes(levelTypes, requiredAccessLevel)) {
+                    throw new Error(`Invalid requiredAccess level. Value provided: ${requiredAccessLevel}. Possible values are: ${levelTypes}.`);
+                }
+                const bindings = allBindings();
+                const bindingsArray = Object.keys(bindings);
+                const requiredAccessBindingLocation = bindingsArray.indexOf("requiredAccess");
+                
+                const activeDatabase = activeDatabaseTracker.default.database();
+                if (activeDatabase) {
+                    if (bindings.visible) {
+                        this.verifyBindingLocation(bindingsArray, "visible", requiredAccessBindingLocation);
+                    }
+                    if (bindings.hidden) {
+                        this.verifyBindingLocation(bindingsArray, "hidden", requiredAccessBindingLocation);
+                    }
+                    if (bindings.disable) {
+                        this.verifyBindingLocation(bindingsArray, "disable", requiredAccessBindingLocation);
+                    }
+                    if (bindings.enable) {
+                        this.verifyBindingLocation(bindingsArray, "enable", requiredAccessBindingLocation);
+                    }
+                } else {
+                    throw new Error("Cannot use the 'requiredAccess' binding - no database is active.");
+                }
+
+                const strategyTypes = ["hide", "disable", "visibilityHidden"];
```

Comment:
maybe let's remove visibilityHidden from this so it will throw and reflect current impl progress ?

Reply:
done